core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -752,23 +751,20 @@ class DAGScheduler(
         submitStage(stage)
       }
     }
-    submitWaitingStages()
   }
 
   /**
    * Check for waiting stages which are now eligible for resubmission.
-   * Ordinarily run on every iteration of the event loop.
+   * Ordinarily run after the parent stage completed successfully.
```

Comment:
Can you update this to say something like:

"Submits stages that depend on the given parent stage. Called when the parent stage completes successfully."